src/main/java/com/unrelentless/fallfest116/mixin/LeavesBlockMixin.java

```diff
@@ -35,21 +35,21 @@ private void injectAppendProperties(StateManager.Builder<Block, BlockState> buil
     @Inject(at = @At("TAIL"), method = "randomTick(Lnet/minecraft/block/BlockState;Lnet/minecraft/server/world/ServerWorld;Lnet/minecraft/util/math/BlockPos;Ljava/util/Random;)V")
     private void injectRandomTick(BlockState state, ServerWorld world, BlockPos pos, Random random, CallbackInfo info) {
         if (state.get(GhostEntity.FALLED) == true) {
-            BlockPos.Mutable mutable = new BlockPos.Mutable();
-            Direction[] var6 = AbstractBlock.FACINGS;
-            int var7 = var6.length;
+            BlockPos.Mutable mutableBlockPos = new BlockPos.Mutable();
+            Direction[] directions = AbstractBlock.FACINGS;
+
+            for (int eachDirection = 0; eachDirection < directions.length; ++eachDirection) {
+                Direction direction = directions[eachDirection];
+                mutableBlockPos.set(pos, direction);
+                BlockState blockState = world.getBlockState(mutableBlockPos);
 
-            for (int var8 = 0; var8 < var7; ++var8) {
-                Direction direction = var6[var8];
-                mutable.set(pos, direction);
-                BlockState blockState = world.getBlockState(mutable);
                 if (blockState.getBlock() instanceof LeavesBlock) {
                     if (!blockState.get(GhostEntity.FALLED)) {
-                        BlockState blockState2 = blockState
-                                .getStateForNeighborUpdate(direction.getOpposite(), state, world, mutable, pos)
+                        BlockState updatedBlockState = blockState
```

Comment:
We can try.
We're getting the neighboring block's state, then appending the "Falled" prop and new random Colour prop to the state.
What would you call that state?

`neighbouringLeavesFalledColourState`?

Reply:
Wow, so many things going on. I can't think of any good names now either. 

The reason why I asked is just that it would be nice to understnad what happens here by looking at the variable name but it probably too difficult for this...